File: LSL/liblsl-Java/src/examples/SendDataInChunks.java
package examples;
import edu.ucsd.sccn.LSL;
import java.io.IOException;

public class SendDataInChunks {
    public static void main(String[] args) throws IOException, InterruptedException  {
        System.out.println("Creating a new StreamInfo...");
        LSL.StreamInfo info = new LSL.StreamInfo("BioSemi","EEG",8,100,LSL.ChannelFormat.float32,"myuid324457");

        System.out.println("Creating an outlet...");
        LSL.StreamOutlet outlet = new LSL.StreamOutlet(info);
        
        System.out.println("Sending data...");
        float[] chunk = new float[10*8];
        for (int t=0;t<100000;t++) {
            // the chunk array contains first all values for the first sample, then the second, and so on
            for (int k=0;k<chunk.length;k++)
                chunk[k] = (float)Math.random()*50-25;
            outlet.push_chunk(chunk); // note: it is also possible to pass in time stamps
            Thread.sleep(100);
        }
        
        outlet.close();
        info.destroy();
    }
}

File: LSL/liblsl-Csharp/examples/SendDataInChunks.cs
using System;
using System.Threading;
using LSL;

namespace ConsoleApplication1
{
    class Program
    {
        static void Main(string[] args)
        {
            Random rnd = new Random();
			// create stream info and outlet
            liblsl.StreamInfo info = new liblsl.StreamInfo("BioSemi", "EEG", 8, 100, liblsl.channel_format_t.cf_float32, "sddsfsdf");
            liblsl.StreamOutlet outlet = new liblsl.StreamOutlet(info);

			// send data in chunks of 10 samples and 8 channels
            float[,] data = new float[10, 8];
            while (true)
            {
                for (int s = 0; s < 10; s++)
                    for (int c = 0; c < 8; c++)
                        data[s, c] = rnd.Next(-100, 100);
                outlet.push_chunk(data);
                System.Threading.Thread.Sleep(100);
            }
        }
    }
}
